containers/src/Kokkos_ScatterView.hpp

```diff
@@ -83,120 +83,126 @@ namespace Experimental {
 template <typename ExecSpace>
 struct DefaultDuplication;
 
-template <typename ExecSpace, int duplication>
+template <typename ExecSpace, typename duplication>
 struct DefaultContribution;
 
 #ifdef KOKKOS_ENABLE_SERIAL
 template <>
 struct DefaultDuplication<Kokkos::Serial> {
-  enum : int { value = Kokkos::Experimental::ScatterNonDuplicated };
+  using duplication_type = Kokkos::Experimental::ScatterNonDuplicated;
```

Comment:
I suggest to use `duplication_tag` rather than `duplication_type` and to prefer the name "tag" when you refer to the new classes you introduced to replace the enum.

I am not firmly asking you to change.  Wondering what others think.
@crtrott @dhollman

Reply:
Ping @crtrott @dhollman

Reply:
I think `duplication_tag` is a better name, as it indicates what it is.

That leads to side questions:

1. Should "Tag" be part of the name of the tag, as in `ScatterNonDuplicatedTag`?  (It seems like extra verbosity which doesn't really convey extra meaning, so I'm leaning against it.)
2. Should tag types have `explicit` default constructors?  When passed to functions, it disallows aggregate initialization `{}` syntax.  I don't think it matters at all in this case, but we should have a consistent policy towards it.

Reply:
I believe I have addressed these issues @dalg24 @nliber please review